### DISCUSSION TITLE
How to horizontal filter a json value?

### DISCUSSION THREAD

**Discussion by dnk8n** [PostgREST/postgrest#2743] [Q&A] [answered]
e.g. A plain get: `GET http://localhost:3000/projects`

```
[
	{
		"id": 3,
		"members": [
			{
				"id": 2,
				"name": "email@example.com"
			}
		],
		"name": "Project Name",
		"activities": [
			{
				"id": 24,
				"name": "Finance"
			}
		],
		"areas_of_work": [
			{
				"id": 4,
				"name": "Observatory"
			}
		]
	}
]
```

I want to filter where areas_of_work.id == 4

I thought it should be `GET http://localhost:3000/projects?areas_of_work-->id=eq.4` but that doesn't work.

Error:
```
{
	"code": "PGRST100",
	"details": "unexpected \"-\" expecting \">\", letter or digit",
	"hint": null,
	"message": "\"failed to parse tree path (areas_of_work-->id)\" (line 1, column 15)"
}
```

**Comment by laurenceisla** [comment, marked as answer]
Hi, it could be done more easily with a `jsonb` column and the [`@>` operator](https://www.postgresql.org/docs/14/functions-json.html#FUNCTIONS-JSONB-OP-TABLE) (`cs` in PostgREST). You could change the column from a `json` to `jsonb` or add a new [computed column](https://postgrest.org/en/latest/api.html#computed-virtual-columns) that casts `areas_of_work`  to `jsonb`.

Then the request would be:

```http
GET /projects?areas_of_work=cs.[{"id":1}]
```
